File: TestApp/app/src/main/java/com/romiope/testapp/database/AppDatabase.java
package com.romiope.testapp.database;

import android.arch.persistence.room.Database;
import android.arch.persistence.room.RoomDatabase;
import com.romiope.testapp.network.entities.ServerResponse;

@Database(entities = {ServerResponse.class}, version = 1)
public abstract class AppDatabase extends RoomDatabase {
    public abstract ServersDao serversDao();
}

File: app/src/main/java/com/example/androidparty/dao/AppDatabase.kt
package com.example.androidparty.dao

import android.content.Context
import androidx.room.Database
import androidx.room.Room
import androidx.room.RoomDatabase
import com.example.androidparty.db.ServerEntity

@Database(entities = [ServerEntity::class], version = 1)
abstract class AppDatabase : RoomDatabase() {
    abstract fun serverDao(): ServerDao

    companion object {
        @Volatile
        private var instance: AppDatabase? = null

        fun getDatabase(context: Context): AppDatabase {
            return instance ?: synchronized(this) {
                val db = Room.databaseBuilder(
                    context.applicationContext,
                    AppDatabase::class.java,
                    "database"
                )
                    .allowMainThreadQueries()
                    .build()
                instance = db
                db
            }
        }
    }
}